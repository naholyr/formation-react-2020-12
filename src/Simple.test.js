import { render, screen } from "@testing-library/react";

const Simple = () => {
  return (
    <p>
      Bonjour, <strong>le monde</strong>.
    </p>
  );
};

test("Jest.expect.toEqual", () => {
  const value = 42;
  expect(value).toEqual(42);
});

test("Simple test has 'le monde'", () => {
  render(<Simple />);
  // https://testing-library.com/docs/dom-testing-library/cheatsheet#queries
  // Kent C. Dodds: "Write tests, not too many, mostly integration"

  // get*    →   un seul élément, erreur si pas trouvé
  // query*  →   un seul élément, pas d'erreur si pas trouvé
  // find*   →   un seul élément, attend si pas trouvé

  // *All*   →   plusieurs éléments

  // *By*    →   type de filtre

  screen.getByText("monde", { exact: false });
  const element = screen.getByText("le monde");
  const p = screen.getByText("Bonjou", { exact: false });
  expect(p).toContainElement(element);
});

test("Simple snapshot", () => {
  const { container } = render(<Simple />);
  expect(container).toMatchSnapshot();
});
