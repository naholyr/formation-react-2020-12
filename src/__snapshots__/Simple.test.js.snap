// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple snapshot 1`] = `
<div>
  <p>
    Bonjour, 
    <strong>
      le monde
    </strong>
    .
  </p>
</div>
`;
